helix-term/src/ui/picker.rs

```diff
@@ -60,14 +70,45 @@ impl<T> FilePicker<T> {
             })
     }
 
-    fn calculate_preview(&mut self, editor: &Editor) {
-        if let Some((path, _line)) = self.current_file(editor) {
-            if !self.preview_cache.contains_key(&path) && editor.document_by_path(&path).is_none() {
-                // TODO: enable syntax highlighting; blocked by async rendering
-                let doc = Document::open(&path, None, Some(&editor.theme), None).unwrap();
-                self.preview_cache.insert(path, doc);
-            }
+    /// Get preview from preivew cache.
+    /// If not cached, calculate and cache preview, then return reference of it.
+    fn get_preview<'a, 'b>(
+        &'a mut self,
+        path: impl AsRef<std::path::Path>,
+        editor: &'b Editor,
+    ) -> &'a Preview {
+        let path = path.as_ref();
+
+        if self.preview_cache.contains_key(path) && editor.document_by_path(path).is_some() {
+            return self.preview_cache.get(path).unwrap();
         }
```

Comment:
A document will never be in both the cache and in the editor, this needs to be

```suggestion
        if let Some(document) = editor.document_by_path(path) {
          return Preview::Document(&document)
        }
        if let Some(preview) = self.preview_cache.get(path) {
            return preview;
        }
```

 but the problem is that `Preview::Document` always has `Document`, not `&Document`. Maybe we need to add another enum variant.

\cc @sudormrfbin

Reply:
Just change && into || will fix?

Reply:
> Just change && into || will fix?

If we do that the unwrap at L83 will be triggered in the case where the document already exists in the editor (so the `||` condition will pass) but we still try to retrieve it from the preview cache.